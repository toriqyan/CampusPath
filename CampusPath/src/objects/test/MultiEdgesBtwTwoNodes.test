# Tests that a graph can be created, three nodes and
# three edges from the same parent to the same child 
# and one edge from the same parent to a different child can be added, 
# and three nodes and four edges can be printed out.

# Create a graph
CreateGraph TheLordOfTheRing3

# Add three nodes
AddNode TheLordOfTheRing3 Frodo
AddNode TheLordOfTheRing3 Gandalf
AddNode TheLordOfTheRing3 Samwise

# Add four edges
AddEdge TheLordOfTheRing3 Frodo Gandalf TheHobbit
AddEdge TheLordOfTheRing3 Frodo Samwise TheReturnOfTheKing
AddEdge TheLordOfTheRing3 Frodo Samwise TheTwoTowers
AddEdge TheLordOfTheRing3 Frodo Samwise TheFellowshipOfTheRing

# Print the nodes in the graph and the outgoing edges from Frodo
ListNodes TheLordOfTheRing3
ListChildren TheLordOfTheRing3 Frodo