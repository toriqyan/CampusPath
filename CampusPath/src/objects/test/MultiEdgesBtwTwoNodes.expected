# Tests that a graph can be created, three nodes and
# three edges from the same parent to the same child 
# and one edge from the same parent to a different child can be added, 
# and three nodes and four edges can be printed out.

# Create a graph
created graph TheLordOfTheRing3

# Add three nodes
added node Frodo to TheLordOfTheRing3
added node Gandalf to TheLordOfTheRing3
added node Samwise to TheLordOfTheRing3

# Add four edges
added edge TheHobbit from Frodo to Gandalf in TheLordOfTheRing3
added edge TheReturnOfTheKing from Frodo to Samwise in TheLordOfTheRing3
added edge TheTwoTowers from Frodo to Samwise in TheLordOfTheRing3
added edge TheFellowshipOfTheRing from Frodo to Samwise in TheLordOfTheRing3

# Print the nodes in the graph and the outgoing edges from Frodo
TheLordOfTheRing3 contains: Frodo Gandalf Samwise
the children of Frodo in TheLordOfTheRing3 are: Gandalf(TheHobbit) Samwise(TheFellowshipOfTheRing) Samwise(TheReturnOfTheKing) Samwise(TheTwoTowers)